app/frontend/src/features/auth/jail/TOSSection.tsx

```diff
@@ -0,0 +1,31 @@
+import React, { useCallback, useState } from "react";
+import Button from "../../../components/Button";
+import TextBody from "../../../components/TextBody";
+import TOS from "../../../components/TOS";
+import { acceptTOS } from "../../../service/jail";
+
+export default function TOSSection({
+  updateJailed,
+}: {
+  updateJailed: () => Promise<any>;
+}) {
+  const [completed, setCompleted] = useState(false);
+  const [loading, setLoading] = useState(false);
+
+  const accept = useCallback(async () => {
+    setLoading(true);
+    const info = await acceptTOS();
+    if (!info.isJailed) await updateJailed();
+    setLoading(false);
+    setCompleted(true);
```

Comment:
Put it in the store.

Reply:
Yeah also I think you can derive this state from `loading` and `jailed` from the auth state in the Redux store?

Reply:
if `jailed` is false, then they must have completed (agreed) to the TOS so I think you can get rid of the `completed` state in this component. Same for the `loading` state since you also have that from the Redux store already.